src/backend/distributed/transaction/transaction_management.c

```diff
@@ -609,6 +628,13 @@ CoordinatedSubTransactionCallback(SubXactEvent event, SubTransactionId subId,
 				CoordinatedRemoteTransactionsSavepointRelease(subId);
 			}
 			PopSubXact(subId);
+
+			/* Set CachedDuringCitusCreation to one level lower to represent citus creation is done */
+
+			if (GetCitusCreationLevel() == subId)
```

Comment:
This seems wrong -- you're comparing a nesting level to a transaction ID.

Reply:
I see why this is wrong but I'm curious to why it worked when I was testing it?